File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        a = dict()

        for i in range(len(nums)):
            a[nums[i]] = i
        
        for i in range(len(nums)):
            if target - nums[i] in a and a[target-nums[i]] != i:
                return [i, a[target - nums[i]]]
        
        return [-1,-1]

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int l = 0;
        int r = nums.length-1;
        int[] res = new int[2];
        while(l < r){
            if(nums[l] + nums[r] < target){l+=1;}
            else if(nums[l] + nums[r] > target){r-=1;}
            else{res[0] = l+1;res[1]=r+1;return res;}
        }
        return res;
    }
    
}
